pwndbg/disasm/__init__.py

```diff
@@ -83,8 +83,16 @@ def get_disassembler(pc):
     extra = None
 
     if pwndbg.arch.current in ('arm', 'aarch64'):
-        extra = {0:CS_MODE_ARM,
-                 0x20:CS_MODE_THUMB}[pwndbg.regs.cpsr & 0x20]
+        if pwndbg.regs.xpsr:
```

Comment:
Is this condition correct? If `xpsr` would have a value of `0` we will assume the `cpsr` shall be used.

This should probably be `if pwndbg.regs.xpsr is not None:`?

Reply:
You are right, value of 0 IS a valid cpsr value and should be considered.